pynpoint/processing/psfpreparation.py

```diff
@@ -123,32 +131,40 @@ def run(self) -> None:
 
         # Run the PSFpreparationModule for each subset of frames
         start_time = time.time()
-        for i, _ in enumerate(frames[:-1]):
+        for j in range(wavelength_dim):
+            for i, _ in enumerate(frames[:-1]):
 
-            # Print progress to command line
-            progress(i, len(frames[:-1]), 'Preparing images for PSF subtraction...', start_time)
+                # Print progress to command line
+                progress(i + j * len(frames[:-1]), wavelength_dim * len(frames[:-1]),
+                         'Preparing images for PSF subtraction...', start_time)
 
-            # Get the images and ensure they have the correct 3D shape with the following
-            # three dimensions: (batch_size, height, width)
-            images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                # Get the images and ensure they have the correct 3D shape with the following
+                # three dimensions: (batch_size, height, width)
+                if len(im_shape) < 4:
+                    images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                else:
+                    images = self.m_image_in_port[j, frames[i]:frames[i+1], ]
 
-            if images.ndim == 2:
-                warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
-                              'An extra dimension has been added.')
+                if images.ndim == 2:
+                    warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
+                                  'An extra dimension has been added.')
 
-                images = images[np.newaxis, ...]
+                    images = images[np.newaxis, ...]
 
-            # Apply the mask, i.e., set all pixels to 0 where the mask is False
-            images[:, ~mask] = 0.
+                # Apply the mask, i.e., set all pixels to 0 where the mask is False
+                images[:, ~mask] = 0.
 
-            # If desired, normalize the images using the Frobenius norm
-            if self.m_norm:
-                im_norm = np.linalg.norm(images, ord='fro', axis=(1, 2))
-                images /= im_norm[:, np.newaxis, np.newaxis]
-                norms.append(im_norm)
+                # If desired, normalize the images using the Frobenius norm
```

Comment:
Is this still correct with 4D data? Perhaps we should make the normalization (which I am anyway not a fan of...) only compatible with 3D data?

Reply:
It applies the nomralization to each wavelength seperatly, so it should be similar to the 3D case. But I agree that only allowing it for 3D data is the best solution.

Reply:
OK let's add a warning in case the normalization is set to True with 4D data, and turn off the normalization in that case.

Reply:
I added the warning and turned off normalization for 4D data.